thriftbp/server_middlewares.go

```diff
@@ -380,3 +383,50 @@ func RecoverPanic(name string, next thrift.TProcessorFunction) thrift.TProcessor
 		},
 	}
 }
+
+// PrometheusMetricMiddleware returns middleware to track Prometheus metrics specific to the Thrift service.
+func PrometheusMetricMiddleware(serviceSlug string) thrift.ProcessorMiddleware {
+	return func(method string, next thrift.TProcessorFunction) thrift.TProcessorFunction {
+		process := func(ctx context.Context, seqID int32, in, out thrift.TProtocol) (success bool, err thrift.TException) {
+			start := time.Now()
+			activeRequestLabels := prometheus.Labels{
+				"thrift_service": serviceSlug,
```

Comment:
@fishy Is there any way to extract the notion of the "thrift service" from the notion of the "server"?  For example, for awhile we were serving the same thrift service from both pref-go and pref-py -- maybe we could just use the struct name for the thrift service name?

Reply:
we don't have a canonical way to do that right now, and I don't think just use the struct name is a good solution to that, but it could potentially be a good default fallback (e.g. when the config didn't specify the name of the service, we extract it from the struct name).